src/app/AttributeAccessInterfaceRegistry.cpp

```diff
@@ -60,43 +58,28 @@ void UnregisterMatchingAttributeAccessInterfaces(F shouldUnregister)
 
 } // namespace
 
-void unregisterAttributeAccessOverride(AttributeAccessInterface * attrOverride)
-{
-    gAttributeAccessInterfaceCache.Invalidate();
-    UnregisterMatchingAttributeAccessInterfaces([attrOverride](AttributeAccessInterface * entry) { return entry == attrOverride; });
-}
+namespace chip {
+namespace app {
 
-void unregisterAllAttributeAccessOverridesForEndpoint(EmberAfDefinedEndpoint * definedEndpoint)
+AttributeAccessInterfaceRegistry & AttributeAccessInterfaceRegistry::Instance()
 {
-    UnregisterMatchingAttributeAccessInterfaces(
-        [endpoint = definedEndpoint->endpoint](AttributeAccessInterface * entry) { return entry->MatchesEndpoint(endpoint); });
+    static AttributeAccessInterfaceRegistry instance;
+    return instance;
 }
 
-bool registerAttributeAccessOverride(AttributeAccessInterface * attrOverride)
+void AttributeAccessInterfaceRegistry::UnregisterAllForEndpoint(EmberAfDefinedEndpoint * definedEndpoint)
```

Comment:
If we are changing this anyway, the argument here should be `EndpointId`.  It was convenient to have it be `EmberAfDefinedEndpoint` back when this all lived in ember stuff, but we don't need that dependency now, right?

Followup is fine for this part.